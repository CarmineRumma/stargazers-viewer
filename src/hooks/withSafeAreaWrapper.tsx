import * as React from 'react';
import {Appearance, SafeAreaView} from 'react-native';
// eslint-disable-next-line @typescript-eslint/no-unused-vars
const colorScheme = Appearance.getColorScheme();

const withSafeArea = (WrappedComponent: React.ComponentType<any>) => {
  return (props: any) => {
    return (
      <SafeAreaView style={{flex: 1}}>
        <WrappedComponent {...props} />
      </SafeAreaView>
    );
  };
};
export default withSafeArea;
