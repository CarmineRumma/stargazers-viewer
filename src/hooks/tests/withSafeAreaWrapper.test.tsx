import React from 'react';
import renderer from 'react-test-renderer';
import {default as withSafeAreaWrapper} from '../withSafeAreaWrapper';
import {View} from 'react-native';

describe('withSafeAreaWrapper', () => {
  it('should render correctly', () => {
    // @ts-ignore
    const {toJSON} = renderer.create(withSafeAreaWrapper(<View />));

    expect(toJSON()).toMatchSnapshot();
  });
});
