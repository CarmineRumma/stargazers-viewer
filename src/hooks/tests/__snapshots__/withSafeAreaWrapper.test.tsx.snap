// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withSafeAreaWrapper should render correctly 1`] = `null`;
